mqtt/src/main/scala/akka/stream/alpakka/mqtt/MqttFlowStage.scala

```diff
@@ -219,8 +219,13 @@ final class MqttFlowStage(sourceSettings: MqttSourceSettings,
             .tryFailure(
               new IllegalStateException("Cannot complete subscription because the stage is about to stop or fail")
             )
-        try mqttClient.disconnect()
-        catch { case _: Throwable => () }
+
+        // Try to disconnect. In case disconnect attempt failed try again with force and throw here if it does not work
+        Try(mqttClient.disconnect().waitForCompletion())
```

Comment:
I'm not sure if blocking with `waitForCompletion` is a good choice here. 
Should it be wrapped in Futures, @johanandren ?

Reply:
Yeah, the stage does not run on the blocking-io-dispatcher, so this should be done in futures, on the blocking-io-dispatcher.